File: contracts/requests/PlaylistRequests.kt
package contracts.requests

import kotlinx.serialization.Serializable

@Serializable
data class CreatePlaylistRequest(
    val title: String,
    val description: String? = null,
    val isPrivate: Boolean = true,
    val initialMediaIds: List<String> = emptyList()
)

@Serializable
data class UpdatePlaylistRequest(
    val playlistId: String,
    val title: String? = null,
    val description: String? = null,
    val isPrivate: Boolean? = null,
    val coverArtUri: String? = null
)

@Serializable
data class DeletePlaylistRequest(
    val playlistId: String
)

@Serializable
data class AddPlaylistItemRequest(
    val playlistId: String,
    val mediaId: String,            // Canonical Media ID
    val targetPositionOrder: Int? = null
)

@Serializable
data class RemovePlaylistItemRequest(
    val playlistId: String,
    val itemId: String
)

@Serializable
data class ReorderPlaylistRequest(
    val playlistId: String,
    val orderedItemIds: List<String>
)

File: contracts/requests/PlaylistRequests.ts
export interface CreatePlaylistRequest {
  title: string;
  description?: string | null;
  isPrivate?: boolean;
  initialMediaIds?: string[];
}

export interface UpdatePlaylistRequest {
  playlistId: string;
  title?: string | null;
  description?: string | null;
  isPrivate?: boolean | null;
  coverArtUri?: string | null;
}

export interface DeletePlaylistRequest {
  playlistId: string;
}

export interface AddPlaylistItemRequest {
  playlistId: string;
  mediaId: string; // Canonical Media ID
  targetPositionOrder?: number | null;
}

export interface RemovePlaylistItemRequest {
  playlistId: string;
  itemId: string;
}

export interface ReorderPlaylistRequest {
  playlistId: string;
  orderedItemIds: string[];
}
